code/game/objects/items.dm

```diff
@@ -119,7 +119,10 @@ GLOBAL_VAR_INIT(rpg_loot_items, FALSE)
 	var/list/attack_verb
 	/// list() of species types, if a species cannot put items in a certain slot, but species type is in list, it will be able to wear that item
 	var/list/species_exception = null
-
+	///A dictionary of a species and a DMI path to use as an alternative sprite for any given item
+	var/list/sprite_sheets = null
+	///A list of species that the item cannot be worn by
+	var/list/species_restricted = null
```

Comment:
I'm also concerned that this means any item with either species restriction or per-species sprite will have an additional list for each of those *for every such item in existence.* I don't know nearly enough about BYOND to be able to tell if this is going to be an issue, but I feel like it might need to be reworked to use a global dictionary based on typepath.

Reply:
> 
> 
> I'm also concerned that this means any item with either species restriction or per-species sprite will have an additional list for each of those _for every such item in existence._ I don't know nearly enough about BYOND to be able to tell if this is going to be an issue, but I feel like it might need to be reworked to use a global dictionary based on typepath.

Works fine for Baystation.

Reply:
> 
> 
> Is it a good idea for this to be a blacklist? Sounds like this is something where the specific species won't be able to wear _most_ items.

Species_restricted will allow for flexible blacklists on either a parent object or specific clothing items. It will likely primarily be used to blacklist large amount of items by inheritance, due to a lack of sprites.

Reply:
it working for baystation is the lowest bar in the world. they barely have any memory or performance concerns due to the lack of gameplay.

Reply:
> 
> 
> it working for baystation is the lowest bar in the world. they barely have any memory or performance concerns due to the lack of gameplay.

If theres a better way to do it then do suggest, this is a draft after all.

Reply:
use string lists

Reply:
**TO DO BASED ON SUGGESTIONS**

- [ ] Look into using an associative string list instead of a dictionary for sprite_sheets
- [ ] Make species_restricted into a bitfield instead, based on bodytype

Reply:
Original author of species_restricted on Bay chipping in to say that aping this setup is a bad idea, you would be better served by redesigning it from scratch. When I coded it I had no idea how much of a memory concern lists were.

Reply:
> 
> 
> Original author of species_restricted on Bay chipping in to say that aping this setup is a bad idea, you would be better served by redesigning it from scratch. When I coded it I had no idea how much of a memory concern lists were.

Completed. 🙏